client/dashboard/sites/deployments/trigger-deployment-modal.tsx

```diff
@@ -0,0 +1,144 @@
+import { createCodeDeploymentRunMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import {
+	Button,
+	Modal,
+	SelectControl,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm, Field, NormalizedField } from '@wordpress/dataviews';
+import { useMemo, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { CodeDeploymentData } from '../../../sites/deployments/deployments/use-code-deployments-query';
+import { ButtonStack } from '../../components/button-stack';
+
+interface TriggerDeploymentModalProps {
+	onClose: () => void;
+	deployments: CodeDeploymentData[];
+}
+
+interface DeploymentFormData {
+	selectedDeploymentId: string;
+}
+
+const form = {
+	layout: { type: 'regular' as const },
+	fields: [ 'selectedDeploymentId' ],
+};
+
+export interface DeploymentSelectControlProps {
+	data: DeploymentFormData;
+	field: NormalizedField< DeploymentFormData >;
+	onChange: ( edits: Partial< DeploymentFormData > ) => void;
+}
+
+function DeploymentSelectControl( { data, field, onChange }: DeploymentSelectControlProps ) {
+	const { id, label, elements, getValue } = field;
+
+	const options = [ { label: __( 'Select a repository' ), value: '' }, ...elements ];
+
+	return (
+		<SelectControl
+			__next40pxDefaultSize
+			label={ label }
+			onChange={ ( value ) => onChange( { [ id ]: value } ) }
+			options={ options }
+			value={ getValue( { item: data } ) }
+		/>
+	);
+}
+
+export function TriggerDeploymentModal( { onClose, deployments }: TriggerDeploymentModalProps ) {
+	const { createSuccessNotice } = useDispatch( noticesStore );
+	const [ deploymentFormData, setDeploymentFormData ] = useState< DeploymentFormData >( {
+		selectedDeploymentId: '',
+	} );
+
+	const selectedDeployment: CodeDeploymentData | null = useMemo( () => {
+		if ( ! deploymentFormData.selectedDeploymentId ) {
+			return null;
+		}
+
+		return (
+			deployments.find(
+				( deployment ) => deployment.id.toString() === deploymentFormData.selectedDeploymentId
+			) || null
+		);
+	}, [ deploymentFormData.selectedDeploymentId, deployments ] );
+
+	const { mutate: mutateCreateCodeDeploymentRun, isPending: isPendingCreateCodeDeploymentRun } =
```

Comment:
You asked for naming guidelines (p1758287566638539/1758192140.079979-slack-CRWCHQGUB) so I'll do it here 🙈 

I think we can come up with a better naming here.`mutateCreateCodeDeploymentRun` sounds like we're mutating `CreateCodeDeploymentRun`, which is not accurate. From what I've seen, people tend to just use the verb, so something like 

```suggestion
	const { mutate: createCodeDeploymentRun, isPending: isCreatingCodeDeploymentRun } =
```

what do you think?

Reply:
Other than that, I think this PR is in a good shape, and if your team has tested it, I think it's good to go!

Reply:
Yeah, I was also considering this name, but the thing which I don't like it's thsi this the name of request function [createCodeDeploymentRun](https://github.com/Automattic/wp-calypso/pull/105782/files#diff-f2b1a161b1fae1c49a8b2101150644cd1943e3a13ac62ca5191c48f54fd0cc37R4) 😅
Anyway I renamed, but it would be cool to have different names to distinguish functions. Dificult case with namings 😅